scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -213,6 +218,17 @@ def handle(self, line):
                 self.recording.append(csv)
 
         self.process_test(line)
+        # Reset the resulting test state to 'failed' for 'one_line' and
+        # ordered 'multi_line' patterns when not all of these patterns were
+        # found in the output, but just ztest's 'PROJECT EXECUTION SUCCESSFUL'.
+        # It might happen because of the pattern sequence diverged from the
+        # test code, the test platform has console issues, or even some other
+        # test image was executed.
+        # TODO: Introduce explicit match policy type either to reject
+        # unexpected console output, or to allow missing patterns.
+        if self.state == "passed" and self.ordered and self.next_pattern < self.patterns_expected:
+            logger.error(f"HARNESS:{self.__class__.__name__}: failed with only {self.next_pattern} matched patterns from expected {self.patterns_expected}")
```

Comment:
Minor issue: should be `{self.next_pattern+1}`

Reply:
it is for `matched patterns`, so without any match it is `with only 0 matched patterns` which sounds a bit odd, I agree

Reply:
I added one line to patterns in `tests/subsys/debug/coredump_backends/debug.coredump.backends.logging` and I received
`HARNESS:Console: failed with only 2 matched patterns from expected 4`
but should be `3 matched patterns from expected 4`, but this is only log :)

Reply:
> I added one line to patterns in `tests/subsys/debug/coredump_backends/debug.coredump.backends.logging` and I received `HARNESS:Console: failed with only 2 matched patterns from expected 4` but should be `3 matched patterns from expected 4`, but this is only log :)

@gchwier, I can't reproduce your results with one line added, but where ?

`ERROR   - HARNESS:Console: failed with only 2 matched patterns from expected 4`
is for
```
tests:                                                                                                                                                                                                                                          
 debug.coredump.backends.logging:                                                                                                                                                                                                          
       regex:                                                                                                                                                                                                                                        
          - "E: #CD:BEGIN#"                                                                                                                                                                                                                           
          - "E: #CD:5([aA])45([0-9a-fA-F]+)"                                                                                                                                                                                                            
          - "FOOBAR"                
          - "E: #CD:END#"                                                                                                                                                                                                                               
```

and with extra line at the end:
```
tests:                                                                                                                                                                                                                                          
 debug.coredump.backends.logging:                                                                                                                                                                                                          
       regex:                                                                                                                                                                                                                                        
          - "E: #CD:BEGIN#"                                                                                                                                                                                                                           
          - "E: #CD:5([aA])45([0-9a-fA-F]+)"                                                                                                                                                                                                            
          - "E: #CD:END#"                                                                                                                                                                                                                               
          - "FOOBAR"                
```
```
ERROR   - HARNESS:Console: failed with only 3 matched patterns from expected 4
```